pyvista/plotting/lookup_table.py

```diff
@@ -0,0 +1,754 @@
+        self.SetNanColor(*Color(value).float_rgba)
+
+    @property
+    def ramp(self) -> str:
+        """Set the shape of the table ramp.
+
+        This attribute is only used when creating custom color maps and will
+        return ``None`` when a color map has been set with
+        :attr:`LookupTable.cmap`. This will clear any existing color map and
+        create new values for the lookup table when set.
+
+        This value may be either ``"s-curve"``, ``"linear"``, or ``"sqrt"``.
+
+        * The default is S-curve, which tails off gradually at either end.
+        * The equation used for ``"s-curve"`` is ``y = (sin((x - 1/2)*pi) +
+          1)/2``, For an S-curve greyscale ramp, you should set
+          :attr:`LookupTable.n_values`` to 402 (which is ``256*pi/2``) to provide
+          room for the tails of the ramp.
+
+        * The equation for the ``"linear"`` is simply ``y = x``.
+        * The equation for the ``"sqrt"`` is ``y = sqrt(x)``.
+
+        Examples
+        --------
+        Show the default s-curve ramp.
+
+        >>> import pyvista as pv
+        >>> lut = pv.LookupTable()
+        >>> lut.hue_range = (0.0, 0.33)
+        >>> lut.ramp = 's-curve'
+        >>> lut.plot()
+
+        Plot the linear ramp.
+
+        >>> import pyvista as pv
+        >>> lut = pv.LookupTable()
+        >>> lut.hue_range = (0.0, 0.33)
+        >>> lut.ramp = 'linear'
+        >>> lut.plot()
+
+        Plot the ``"sqrt"`` ramp.
+
+        >>> import pyvista as pv
+        >>> lut = pv.LookupTable()
+        >>> lut.hue_range = (0.0, 0.33)
+        >>> lut.ramp = 'sqrt'
+        >>> lut.plot()
+
+        """
+        return RAMP_MAP[self.GetRamp()]
+
+    @ramp.setter
+    def ramp(self, value: str):
+        try:
+            self.SetRamp(RAMP_MAP_INV[value])
+        except KeyError:
+            raise ValueError(f'`ramp` must be one of the following:\n{list(RAMP_MAP_INV.keys())}')
+        self._rebuild()
+
+    @property
+    def above_range_color(self) -> Optional[Color]:
+        """Return or set the above range color.
+
+        Any values above :attr:`LookupTable.table_range` will be rendered with this
+        color.
+
+        Examples
+        --------
+        Enable the usage of the above range color.
+
+        >>> import pyvista as pv
+        >>> lut = pv.LookupTable()
+        >>> lut.above_range_color = 'blue'
+        >>> lut.plot()
+
+        Disable the usage of the above range color.
+
+        >>> import pyvista as pv
+        >>> lut = pv.LookupTable()
+        >>> lut.above_range_color = None
+        >>> lut.plot()
+
+        """
+        if self.GetUseAboveRangeColor():
+            return Color(self.GetAboveRangeColor())
+        return None
+
+    @above_range_color.setter
+    def above_range_color(self, value):
+        if value is None:
+            self.SetUseAboveRangeColor(False)
+        else:
+            self.SetAboveRangeColor(*Color(value).float_rgba)
+            self.SetUseAboveRangeColor(True)
```

Comment:
What happens if I do something logical like: `lut.above_range_color = True` considering that `color=True` is a valid argument for the `add_mesh` method, there is precedent for this. I'd suggest `True` being a special case that either uses the default color or whatever was last set (default color being pulled from the active theme)

Reply:
Similarly for all like methods

Reply:
Agreed.